src/CodeViewer/include/CodeViewer/Viewer.h

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CODE_VIEWER_VIEWER_H_
+#define CODE_VIEWER_VIEWER_H_
+
+#include <QPlainTextEdit>
+#include <QPointer>
+#include <QResizeEvent>
+#include <QWheelEvent>
+#include <functional>
+
+#include "Em.h"
+#include "PlaceHolderWidget.h"
+
+namespace orbit_code_viewer {
+
+/*
+  Viewer is a for displaying source code. It derives from a QPlainTextEdit
+  and adds some additional features like a left sidebar for displaying
+  line numbers and a heatmap indicator.
+
+  Example usage:
+  Viewer viewer{};
+  viewer.SetEnableLineNumbers(true);
+
+  // Add some spacing left and right of the line number
+  viewer.SetLineNumbersMargins(Em{1.0}, Em{0.3});
+
+  // The heatmap indicator area will be 2em wide.
+  viewer.SetHeatmapBarWidth(Em{2.0});
+
+  viewer.SetHeatmapSource([](int line_number) {
+    (void) line_number;
+
+    // The number we return need to be between 0.0f and 1.0f.
+    return 0.5f;
+  });
+
+  viewer.show();
+
+  Also check out the documentation of QPlainTextEdit for more details on how to
```

Comment:
Only for curiosity. In these cases, shall we have to add a link to the documentation?

Reply:
I thought about that and decided not to, because it will be outdated when we switch Qt versions and we will probably forget about updating comments.

At the same time, finding the docs is pretty easy. A google search for QPlainTextEdit will guide you there in seconds. WDYT?